### PR TITLE
Worked issue #1058. Added a new toast for a situation when history is not present.

### LINKED ISSUE #1058: History is cleared even when no history is present
**Describe the bug**
In history activity, even when no history is present if the user selects clear no history, the Toast says *All history cleared*.

**Expected behavior**
The Toast should be **No history present**.

**Steps to reproduce the behavior:**
1. Open history and select Clear History from the overflow menu.
2. Once again, select Clear history.
3. Toast says **All history cleared**.

**Screenshots**
clearhistory

**Environment**
- Version of Kiwix Android : 2.4 Build 55
- Device : Redmi 4
- OS version : Android Nougat 7.1

### PR DESCRIPTION
**Pull request by devanshi7799**
Fixes #1058 

Here, I have added a new toast for a situation when the history is not present. Guide me if some more changes are needed to be added.

ezgif com-video-to-gif (2)

**Screenshots**

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java [hunk 1]
```diff
@@ -209,12 +209,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
     } else if (itemId == R.id.menu_history_clear) {
       presenter.deleteHistory(new ArrayList<>(fullHistory));
-      fullHistory.clear();
-      historyList.clear();
-      historyAdapter.notifyDataSetChanged();
-      setResult(RESULT_OK, new Intent().putExtra(USER_CLEARED_HISTORY, true));
-      Toast.makeText(this, R.string.all_history_cleared_toast, Toast.LENGTH_SHORT).show();
-      return true;
+
+      if (fullHistory.isEmpty() & historyList.isEmpty()) {
```
Comment (abdulwd) [inline core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java]: Prefer `&&` instead of `&`.

**Comment by iadeelzafar** [comment]
I have two suggestions here,

1- @devanshi7799 can you please add a simple Empty View too? 
Once the history is cleared. there should be an empty view saying

"No History Present"

2- And then we can simply hide the menu button "Clear History"  if there is no History present. 

@abdulwd what do you think?

File: core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java [hunk 2]
```diff
@@ -209,12 +209,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
       presenter.deleteHistory(new ArrayList<>(fullHistory));
-      fullHistory.clear();
-      historyList.clear();
-      historyAdapter.notifyDataSetChanged();
-      setResult(RESULT_OK, new Intent().putExtra(USER_CLEARED_HISTORY, true));
-      Toast.makeText(this, R.string.all_history_cleared_toast, Toast.LENGTH_SHORT).show();
-      return true;
+
+      if (fullHistory.isEmpty() && historyList.isEmpty()) {
+        Toast.makeText(this, "No history present", Toast.LENGTH_SHORT).show();
```
Comment (iadeelzafar) [inline core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java]: Add this string in resource file.

**Comment by devanshi7799** [comment]
@iadeelzafar yes sure. :)

File: core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java (hunk 2 above)
Comment (devanshi7799) [reply core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java]: which resource file?

**Comment by devanshi7799** [comment]
@iadeelzafar can you guide me and tell me why the checks are failing?

**Comment by iadeelzafar** [comment]
@devanshi7799 your build is failing due to Codecov. It checks your changes that either it will reduce or increase code coverage with respect to factors like unit testing, etc.

Since it's a relatively simpler PR, I don't think you need to worry about that.

**Comment by iadeelzafar** [comment]
@devanshi7799 would you like to work on the changes I proposed above? Please let me know if there's anything you need to know more about it.

**Comment by devanshi7799** [comment]
@iadeelzafar Yes, I would like to work on the issues to provided above.
I have made some changes and disabled the clear history option when no issue is present.

**Comment by iadeelzafar** [comment]
@devanshi7799 you need to add an empty view.

**Comment by kelson42** [comment]
@devanshi7799 Do you still plan to work on this PR?

**Comment by devanshi7799** [comment]
@kelson42 i am unable to solve it further. 
anyone else who wants to work on it is welcome.

**Comment by kelson42** [comment]
@iadeelzafar Would you be able to just finish it?

**Comment by macgills** [comment]
also this is not what the ticket wanted, [this](https://github.com/kiwix/kiwix-android/issues/1058#issuecomment-532585456) is what the decision on behaviour was

**Comment by macgills** [comment]
@iadeelzafar if you want to continue this pr please re-open it or start anew, I am now closing this due to a lack of activity

**Pull request closed without merging**